src/models/BanList.ts

```diff
@@ -182,6 +182,38 @@ class BanList extends EventEmitter {
         return [...this.serverRules, ...this.userRules, ...this.roomRules];
     }
 
+    /**
+     * Return all of the rules in this list that will match the provided entity.
+     * If the entity is a user, then we match the domain part against server rules too.
+     * @param ruleKind The type of rule for the entity e.g. `RULE_USER`.
```

Comment:
Could you turn this into a type, to avoid errors in client code?

e.g.
```typescript
export type RuleType = RULE_USER | ...
```

Reply:
It doesn't seem right, it shouldn't be restricted to these rule types.